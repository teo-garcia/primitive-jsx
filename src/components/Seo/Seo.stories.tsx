import * as React from 'react';
import { Meta, Story } from '@storybook/react';
import { SeoProps } from '../../utils/types';
import { Seo } from './Seo';

export default {
  title: 'Seo',
  component: Seo,
} as Meta;

const Template: Story<SeoProps> = (args) => <Seo {...args} />;

export const Default = Template.bind({});

Default.args = {
  title: 'Example | Home',
  description: 'Perfect example for the Seo component',
};
