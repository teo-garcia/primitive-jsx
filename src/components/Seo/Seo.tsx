import * as React from 'react';
import { SeoProps } from '../../utils/types';

function Seo(props: SeoProps) {
  const {
    title,
    description = '',
    ogTitle = '',
    ogDescription = '',
    ogImage = '',
  } = props;
  return (
    <React.Fragment>
      {/* <Head> */}
      <title>{title}</title>
      <meta name="description" content={description} />
      <meta name="viewport" content="width=device-width, initial-scale=1" />
      <meta property="og:title" content={ogTitle} />
      <meta property="og:description" content={ogTitle} />
      <meta property="og:description" content={ogDescription} />
      <meta property="og:image" content={ogImage} />
      {/* </Head> */}
    </React.Fragment>
  );
}

export { Seo };
