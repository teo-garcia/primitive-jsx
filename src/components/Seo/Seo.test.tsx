import * as React from 'react';
import { render } from '@testing-library/react';
import { Seo } from './Seo';

describe('<Layout /> tests', function () {
  test('Should match with the component snapshot', function () {
    const { asFragment } = render(<Seo title="Hello" description="World" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
