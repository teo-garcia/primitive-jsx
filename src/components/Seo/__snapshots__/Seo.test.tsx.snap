// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> tests Should match with the component snapshot 1`] = `
<DocumentFragment>
  <title>
    Hello
  </title>
  <meta
    content="World"
    name="description"
  />
  <meta
    content="width=device-width, initial-scale=1"
    name="viewport"
  />
  <meta
    content=""
    property="og:title"
  />
  <meta
    content=""
    property="og:description"
  />
  <meta
    content=""
    property="og:description"
  />
  <meta
    content=""
    property="og:image"
  />
</DocumentFragment>
`;
